Unknown flags are not validated causing misleading ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL error

### Verify latest release

- [x] I verified that the issue exists in the latest pnpm release

### pnpm version

_No response_

### Which area(s) of pnpm are affected? (leave empty if unsure)

CLI

### Link to the code that reproduces this issue or a replay of the bug

_No response_

### Reproduction steps

1. Setup monorepo project structure
2. Attempt to run `pnpm --filter <package_selector> <command>` with a misspelled `--filter` flag, e.g., `--fitler` or `--xyzabc`
3. Observe output

```bash
> pnpm --fitler web dev    
'web' is not recognized as an internal or external command,
operable program or batch file.
 ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  Command "web" not found
```

### Describe the Bug

pnpm does not fail fast when an unknown top‑level flag is provided (e.g. --fitler instead of --filter). Instead pnpm silently ignores the invalid flag and interprets the next argument as a command (e.g. web). This then reports a misleading error that obscures the real root cause (an invalid flag) and can make diagnosing the issue confusing, especially with minor misspellings.

### Expected Behavior

The expected behavior is to error explicitly on the unknown flag with a clear validation message.

Please note I am referencing from https://pnpm.io/pnpm-cli

> Unlike npm, pnpm validates all options.

### Which Node.js version are you using?

24.12.0

### Which operating systems have you used?

- [ ] macOS
- [x] Windows
- [ ] Linux

### If your OS is a Linux based, which one it is? (Include the version if relevant)

_No response_

--- a/.changeset/curly-panthers-check.md
+++ b/.changeset/curly-panthers-check.md
@@ -1,0 +1,5 @@
+---
+"pnpm": patch
+---
+
+Report unknown top-level options before falling back to implicit `pnpm run` scripts.

--- a/pnpm/src/main.ts
+++ b/pnpm/src/main.ts
@@ -75,7 +75,7 @@ export async function main (inputArgv: string[]): Promise<void> {
     return
   }
 
-  if (unknownOptions.size > 0 && !fallbackCommandUsed && !(cmd && NOT_IMPLEMENTED_COMMAND_SET.has(cmd))) {
+  if (unknownOptions.size > 0 && !(cmd && NOT_IMPLEMENTED_COMMAND_SET.has(cmd))) {
     printError(formatUnknownOptionsError(unknownOptions), `For help, run: pnpm help${cmd ? ` ${cmd}` : ''}`)
     process.exitCode = 1
     return

--- a/pnpm/test/cli.ts
+++ b/pnpm/test/cli.ts
@@ -95,11 +95,11 @@ test('implicit run command fails when an unsupported top-level flag is used', ()
     },
   })
 
-  const { status, stdout, stderr } = execPnpmSync(['--fitler', 'web', 'dev'])
+  const { status, stdout, stderr } = execPnpmSync(['--fitler', 'web', 'dev']) // cspell:disable-line
 
   expect(status).toBe(1)
   expect(stdout.toString()).not.toContain('script should not run')
-  expect(stderr.toString()).toMatch(/Unknown option: 'fitler'/)
+  expect(stderr.toString()).toMatch(/Unknown option: 'fitler'/) // cspell:disable-line
   expect(stderr.toString()).toMatch(/Did you mean 'filter'/)
 })
 

--- a/pnpm/test/cli.ts
+++ b/pnpm/test/cli.ts
@@ -88,6 +88,21 @@ test('command fails when an unsupported flag is used', async () => {
   expect(stderr.toString()).toMatch(/Unknown option: 'save-dev'/)
 })
 
+test('implicit run command fails when an unsupported top-level flag is used', () => {
+  prepare({
+    scripts: {
+      web: 'node -e "console.log(\'script should not run\')"',
+    },
+  })
+
+  const { status, stdout, stderr } = execPnpmSync(['--fitler', 'web', 'dev'])
+
+  expect(status).toBe(1)
+  expect(stdout.toString()).not.toContain('script should not run')
+  expect(stderr.toString()).toMatch(/Unknown option: 'fitler'/)
+  expect(stderr.toString()).toMatch(/Did you mean 'filter'/)
+})
+
 test('command fails when a deprecated option is used', async () => {
   prepare()
 

